src/System.Private.CoreLib/shared/System/Text/StringBuilder.cs

```diff
@@ -2006,23 +2141,35 @@ unsafe private static void ThreadSafeCopy(char[] source, int sourceIndex, char*
             }
         }
 
-        private static void ThreadSafeCopy(char[] source, int sourceIndex, char[] destination, int destinationIndex, int count)
+        // Copies the source StringBuilder to the destination IntPtr memory allocated with len bytes.
+        internal unsafe void InternalCopy(IntPtr dest, int len)
```

Comment:
I'll make these changes in another PR, If I change anything in this one the mirror will break.